src/PrMerger.js

```diff
@@ -43,12 +44,31 @@ class PrMerger {
     // there is a PR still-in-merge.
     async runStep() {
         Logger.info("runStep running");
-        const currentContext = await this._current();
-        if (currentContext && !(await this._finishContext(currentContext)))
-            return true; // still in-process
 
-        const prList = await this._getPRList();
-        this._logPRList(prList);
+        let prList = await this._getPRList();
+
+        let currentContext = await this._current();
+
+        if (currentContext && Config.guardedRun()) {
+            const clearedForMerge = await this._clearedForMerge(currentContext.number());
+            if (!clearedForMerge && prList.length && prList[0].clearedForMerge) {
+                Logger.info("will try to give up the not-cleared-for-merge PR" + currentContext.number() +
```

Comment:
> Why such an uncertainty?

This uncertainty is needed because there is a chance when this attempt (of giving up) was not successful and we are still working with the unlabeled PR. This situation can happen when all labeled PRs are defective.

> It do not think this code should know about labeled PR processing order.

OK.